apisix/plugins/jwe-decrypt.lua

```diff
@@ -0,0 +1,234 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core            = require("apisix.core")
+local consumer_mod    = require("apisix.consumer")
+local base64          = require("ngx.base64")
+local ngx             = ngx
+local sub_str         = string.sub
+local plugin_name     = "jwe-decrypt"
+local cipher          = require("resty.openssl.cipher").new("aes-256-gcm")
+
+local schema          = {
+  type = "object",
+  properties = {
+    header = {
+      type = "string",
+      default = "Authorization"
+    },
+    forward_header = {
+      type = "string",
+      default = "Authorization"
+    },
+    strict = {
+      type = "boolean",
+      default = true
+    }
+  },
+}
+
+local consumer_schema = {
+  type = "object",
+  -- can't use additionalProperties with dependencies
+  properties = {
+    key = { type = "string" },
+    secret = { type = "string", minLength = 32 },
+    base64_secret = { type = "boolean" },
+  },
+  required = { "key", "secret" },
+}
+
+
+local _M = {
+  version = 0.1,
+  priority = 2509,
+  type = 'auth',
+  name = plugin_name,
+  schema = schema,
+  consumer_schema = consumer_schema
+}
+
+
+function _M.check_schema(conf, schema_type)
+  core.log.info("input conf: ", core.json.delay_encode(conf))
+
+  local ok, err
+  if schema_type == core.schema.TYPE_CONSUMER then
+    ok, err = core.schema.check(consumer_schema, conf)
+  else
+    return core.schema.check(schema, conf)
```

Comment:
It is very strange here, how about `ok, err = core.schema.check(schema, conf)`

Reply:
why? in rest of the places apisix follows the same way.